### DISCUSSION TITLE
'Export code' button is exporting GrapesJS boilerplate

### DISCUSSION THREAD

**Discussion by wunksert** [GrapesJS/grapesjs#5104] [General]
Any idea why this might be? For context, I've embedded my grapesJS app into a web framework called Remix, but it's not doing anything except initializing grapes client-side in a useEffect hook. The drag & drop is not working either. There are some additional iframes and scripts, is it possible grapes is getting confused?

<img width="860" alt="Screenshot 2023-05-10 at 8 52 22 PM" src="https://github.com/GrapesJS/grapesjs/assets/12617956/fca732b2-3cd1-4a6c-a207-e57f6533c40d">

**Comment by artf** [comment]
Well, that's weird. Are you able to create a reproducible demo for that (eg. using codesandbox/stackblitz)? Does it work properly without Remix?

**Reply by wunksert** [reply]
Sure I can share a reproducible demo today. It does seem to work without remix. I had to put grapejs.init initialization into a useEffect to work with remix's SSR but that's all I can think is different. Will share demos asap

**Reply by artf** [reply]
so it works properly inside useEffect? Because in that case that's how it should be done (eg. also in React)

**Reply by wunksert** [reply]
No the useEffect in remix is where it breaks. There are a number of issues like the drag and drop not working, a view.getBody call returning null in grapesJs core, and the plugins not loading. When I share a demo with you in a few hours I'll recreate a clean remix instance with a clean grapes instance and that should give some hints as well

**Reply by wunksert** [reply]
@artf here's a link to a reproducible demo: https://github.com/wunksert/remix-grapes-error/tree/main
I included some details in the Readme in case you're not familiar with remix. Let me know if you have any questions

**Reply by wunksert** [reply]
The async dynamic import is required because some of the grapes plugins I want to use initialize something on import using client-only APIs, so they can't be imported statically (or they'll fail during SSR). I tried removing all the plugins and importing the init function statically, so I could remove the async function but that didn't change anything

**Comment by wunksert** [comment]
After some further investigation it's too soon to say the drag and drop functionality isn't working. The 3 blocks present are buggy and don't render anything, that's expected. I need to get the plugins working to properly test the drag and drop

**Comment by wunksert** [comment]
@artf wondered if you've had a chance to look at this any further? Happy to provide any info you need
